src/atomicfile.rs

```diff
@@ -67,7 +84,7 @@ impl<'l> AtomicCreateFile<'l> {
                 // stale. Failure to grab the lock here should be a rare race
                 // condition, but some other process will have the lock and
                 // proceed.
-                file.try_lock_exclusive()?;
+                lock_name(dest, &file)?;
```

Comment:
This returns `WouldBlock` for this rare race condition, in which the file created by this process (Process 1) was `unlink()`-ed by another process (Process 2) before this locking happens (because the file has `len == 0 -> stale`). 

After process 2 unlinks the file, it will attempt to recreate it, lock it and write to it. But does it make sense for process 1 to wait for process 2 to create and lock the file and is that desirable?

Reply:
I'm not sure I follow. `WouldBlock` is treated as an unrecoverable error ("the race is lost, give up"). It does not block in practice. Does this fact change your parse of the situation? The idea I have in mind is that we're trying to open+lock the file, and if any of that fails, then this process doesn't have ownership over the file.